bentoml/adapters/base_output.py

```diff
@@ -32,46 +37,39 @@ def __init__(self, cors='*'):
     def config(self):
         return dict(cors=self.cors,)
 
-    def to_response(self, result, request: flask.Request) -> flask.Response:
-        """Converts corresponding data into an HTTP response
-
-        :param result: result of user API function
-        :param request: request object
+    @property
+    def pip_dependencies(self):
```

Comment:
would be great to also adopt the `set_dependencies(self, env)` approach in the new Artifacts API, e.g.: https://github.com/bentoml/BentoML/blob/master/bentoml/artifact/coreml_model_artifact.py#L120